frontend/src/lib/utils.tsx

```diff
@@ -535,3 +535,23 @@ export function sampleSingle<T>(items: T[]): T[] {
     if (!items.length) throw Error('Items array is empty!')
     return [items[Math.floor(Math.random() * items.length)]]
 }
+
+export function ellipsis(input: string, max: number): string {
+    return input.length > max ? input.substr(0, max) + '...' : input
+}
+
+export function parseGithubRepoURL(url: string): Record<string, string> {
+    const match = url.match(/https?:\/\/(www\.|)github.com\/([^\/]+)\/([^\/]+)\/?$/)
```

Comment:
I think a more refined version of this regex would be `^https?:\/\/(?:www\.)?github\.com\/([A-Za-z0-9_.-]+)\/([A-Za-z0-9_.-]+)\/?$`
Since this has `$` at the end, it could also maybe use `^` at the beginning.
`(www\.|)` seems a bit unclear to me, is there an advantage to this as compared to `(?:www\.)?`
The dot in `github.com` needs to be escaped as well.
GitHub names can only contain alphanumeric chars with dots, underscores and hyphens.

Reply:
I changed it to your suggestion. Good points!

For `(www\.|)` vs `(?:www\.)?` dunno. I find the first form clearer (read as `"www."` or `""`), but it's a total nit.